docs/DeveloperGuide.md

```diff
@@ -18,12 +44,162 @@
+- `Pet#setAge()` - Sets the age of the pet
+- `Pet#setPetType()` - Sets the type of the pet (e.g. Dog, Cat, Parrot)
+- `Pet#setWeight()` - Sets the weight of the pet
+
+These operations are exposed in the `PetList` class as `PetList#addStat()` and `AddStatCommand` class as `AddStatCommand#execute()`.
+
+<p align="center">
+    <img src="images/AddPetStatFeatureClassDiagram.png">
+    <br />
+    <i>Figure 4: Class Diagram for Add Pet Stat Feature</i>
+</p>
+
+Given below is an example usage scenario and how the add pet stat mechanism behaves.
+
+Step 1. After the user launches the application and added a pet named "Bob", a `Pet` object will be initialised and saved in the `petList`.
+
+<p align="center">
+    <img src="images/AddPetStatFeatureObjectDiagram (1).png">
+    <br />
+    <i>Figure 5: Object Diagram for Add Pet Stat Feature After Step 1</i>
+</p>
+
+Step 2. The user executes `add-stat Bob weight 5` command to add a weight stat of 5kg in the `Pet` object. The `add-stat` command calls `AddStatCommand#execute()` then `PetList#addStat()`, causing the `Pet` object's `weight` variable to be modified and saved.
+
+<p align="center">
+    <img src="images/AddPetStatFeatureObjectDiagram (2).png">
+    <br />
+    <i>Figure 6: Object Diagram for Add Pet Stat Feature After Step 2</i>
+</p>
+
+### Remove Pet Stat Command
+
+The `RemoveStatCommand` class inherits its properties from the abstract `Command` class.
+
+This class is executed whenever the `remove-stat <pet name> <stat name>` command is called by the user. The intended functionality of this command is to remove the `<stat name>` from the pet with name `<pet name>`. 
+
+It works like this:
+
+1. The `main()` method calls `ui.getUserInput()` from `ui`. The user's command is parsed through `commandParser.parseCommand()`.
+
+2. Within `commandParser.parseCommand()`, `newCommand()` will identify the command received (`remove-stat`).
+
+3. `RemoveStatCommand.parseArgs()` separates the arguments into the pet name and stat name.
+
+4. `RemoveStatCommand.execute()` calls `PetList.removeStat()` with the pet name and stat name, and calls `ui.removeStatCommandMessage()` to send output to the user.
+
+5. `PetList.removeStat()` uses its internal `find()` method to find the corresponding Pet object, and calls `Pet.removeStat()` on that object.
+
+6. `Pet.removeStat()` uses a switch statement on the stat name to identify the proper method to call: either `setPetType("")`, `setAge("")`, or `setWeight("")`. If the stat name does not match any of those, it prints out an error message.
+
+7. `Pet.removeStat()`, `PetList.removeStat()`, and `RemoveStatCommand` all return, allowing the program to receive another command.
+
+This feature was implemented like this in order to maximize use of the OOP paradigm. 
+
+<p align="center">
+    <img src="images/RemoveStatCommand.png">
+    <br />
+    <i>Figure 7: Sequence Diagram for Remove Pet Stat Command</i>
```

Comment:
Maybe could consider labelling the arrow from PetList to Pet with the function name removeStat()